### ISSUE TITLE
Golem creates 20 subtasks instead of 1

### ISSUE THREAD

**Issue by ederenn** [golemfactory/clay#4362]
## Description

**Golem Version**: 2019-06-18 14:11:06 INFO    GOLEM Version: 0.19.2+dev592.gfe81243
            

**Golem-Messages version** (leave empty if unsure):  golem_messages Version: 3.7.0

**Electron version** (if used): 0.2.0

**OS** [e.g. Windows 10 Pro]: Windows 10 Pro, Linux 18.04

**Branch** (if launched from source):

**Mainnet/Testnet**: mainnet

**Priority label is set to the lowest by default. To setup higher priority please change the label**
_P0 label is set for Severity-Critical/Effort-easy
P1 label is set  for Severity-Critical/Effort-hard
P2 label is set for Severity-Low/ Effort-easy
P3 label is set for Severity-Low/Effort-hard_

**Description of the issue**:

When creating a clasroom task with 1 frame, 1 subtask Golem sometimes creates a task with 20 subtasks shown in task details. On main task list Golem shows 0 subtasks.

**Actual result**:

Golem computes 20 subtasks instead of 1. 
**Screenshots**:

![Screenshot from 2019-06-18 20-06-56](https://user-images.githubusercontent.com/32061891/59708384-f5a0ee80-9204-11e9-8054-1b33b6754d5c.png)

## Steps To Reproduce
_Short description of steps to reproduce the behavior:_
e.g.
1. Launch Golem
2. Create a task using classroom file
3. Set resolution 400x400, samples 30, 1 frame, 1 subtask, task timeout 30 min, subtask timeout 30 min. bid 0.2
4. Check task details if there is one subtask created or 20

error is reproducible but non deterministic, doesn't repeat every try

## Expected behavior

Golem should compute 1 frame with one subtask

## Logs and any additional context

```
C:\Users\ederenn>golemcli --mainnet tasks show
id                                    ETA        subtasks_count  status    completion
------------------------------------  -------  ----------------  --------  ------------
0a89fdf6-9104-11e9-bdd6-b869b4e7c543  0:00:00                 4  Timeout   100.00 %
179b4e3a-910a-11e9-a33c-b869b4e7c543  8:13:24                 4  Restart   75.00 %
29f0efc6-9100-11e9-b5d6-b869b4e7c543  0:00:00                 4  Finished  100.00 %
69ce06a8-910c-11e9-ad54-b869b4e7c543  0:00:00                 4  Finished  100.00 %
9c7b56cc-9112-11e9-a119-b869b4e7c543  0:00:00                 4  Restart   100.00 %
a7564e3e-8ea2-11e9-83dc-b869b4e7c543  0:00:00                 4  Restart   100.00 %
b4dec580-9103-11e9-acd4-b869b4e7c543  0:00:00                 4  Finished  100.00 %
b7e16240-90f2-11e9-9357-b869b4e7c543  0:00:00                 4  Finished  100.00 %
bd17f808-9116-11e9-bec1-b869b4e7c543  0:00:00                 4  Finished  100.00 %
fc62d2a8-9115-11e9-9fd5-b869b4e7c543  ???                     4  Restart   0.00 %
1ef1c806-91c4-11e9-b274-b869b4e7c543  0:00:00                30  Finished  100.00 %
cd8d1692-91d0-11e9-b537-b869b4e7c543  0:00:00                 4  Finished  100.00 %
ed7acf00-91d0-11e9-89ba-b869b4e7c543  0:00:00                 1  Finished  100.00 %
e806d9b4-91d5-11e9-a051-b869b4e7c543  0:00:00                 0  Finished  100.00 %

C:\Users\ederenn>golemcli --mainnet tasks subtasks e806d9b4-91d5-11e9-a051-b869b4e7c543
node               id                                    status     completion
-----------------  ------------------------------------  ---------  ------------
oncoming storm     f815842e-91d5-11e9-b244-b869b4e7c543  Cancelled  0.00 %
rqc1               f81a60c6-91d5-11e9-80a3-b869b4e7c543  Finished   100.00 %
mf/ubuntu.mainnet  f8263458-91d5-11e9-8c30-b869b4e7c543  Cancelled  0.00 %
vm linux           f82fbe66-91d5-11e9-9170-b869b4e7c543  Finished   100.00 %
rqc1               24799866-91d6-11e9-9f52-b869b4e7c543  Finished   100.00 %
vm linux           358fa4e4-91d6-11e9-8ef5-b869b4e7c543  Finished   100.00 %
rqc1               50955526-91d6-11e9-8e27-b869b4e7c543  Finished   100.00 %
vm linux           5d88340c-91d6-11e9-886d-b869b4e7c543  Finished   100.00 %
rqc1               79201f12-91d6-11e9-834a-b869b4e7c543  Finished   100.00 %
vm linux           8862149c-91d6-11e9-b275-b869b4e7c543  Finished   100.00 %
rqc1               9d7cd6c8-91d6-11e9-aadd-b869b4e7c543  Finished   100.00 %
vm linux           b2da05b6-91d6-11e9-898a-b869b4e7c543  Finished   100.00 %
rqc1               c49bf538-91d6-11e9-a866-b869b4e7c543  Finished   100.00 %
vm linux           def014ae-91d6-11e9-8ee8-b869b4e7c543  Finished   100.00 %
rqc1               ea31ff5c-91d6-11e9-b33f-b869b4e7c543  Finished   100.00 %
vm linux           0890f6c6-91d7-11e9-a3d1-b869b4e7c543  Finished   100.00 %
rqc1               176f139e-91d7-11e9-8e69-b869b4e7c543  Finished   100.00 %
rqc1               32fccc0a-91d7-11e9-9e63-b869b4e7c543  Finished   100.00 %
vm linux           3308d55c-91d7-11e9-aee1-b869b4e7c543  Finished   100.00 %
rqc1               55a9759e-91d7-11e9-aa40-b869b4e7c543  Finished   100.00 %
vm linux           658bda6e-91d7-11e9-ace1-b869b4e7c543  Finished   100.00 %
rqc1               80926cd4-91d7-11e9-bba5-b869b4e7c543  Finished   100.00 %

C:\Users\ederenn>
```
```
C:\Users\ederenn>golemcli --mainnet tasks dump e806d9b4-91d5-11e9-a051-b869b4e7c543
{
    "preview": "C:\\Users\\ederenn\\AppData\\Local\\golem\\golem\\default\\mainnet\\ComputerRes\\e806d9b4-91d5-11e9-a051-b869b4e7c543\\tmp\\current_task_preview.PNG",
    "id": "e806d9b4-91d5-11e9-a051-b869b4e7c543",
    "name": "classroom",
    "type": "Blender",
    "subtasks_count": 0,
    "progress": 1.0,
    "options": {
        "frame_count": 1,
        "output_path": "C:\\Users\\ederenn\\Documents\\classroom_2019-06-18_16-32-31",
        "format": "PNG",
        "resolution": [
            400,
            400
        ],
        "frames": "1",
        "compositing": false,
        "samples": 300
    },
    "time_started": 1560868351.9715579,
    "time_remaining": 0.0,
    "last_updated": 1560869075.1060095,
    "status": "Finished",
    "estimated_cost": "2000000000000000000",
    "estimated_fee": "11200000000000045",
    "compute_on": "cpu",
    "timeout": "0:30:00",
    "subtask_timeout": "0:30:00",
    "bid": 0.2,
    "resources": [
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\officeSupplies\\textures\\eraser.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_brightWood.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_frostedGlass.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_08.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\officeSupplies\\textures\\pencil_brandMask.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_06.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\books\\textures\\zapBook.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\dustBin\\textures\\dustbin_wireframe.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\officeSupplies\\textures\\pencil_color.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\wallClock\\wallClock.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\wastes\\textures\\crinkledPaper.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\__ENV\\Garage\\Garage.hdr",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\ReadMe.txt",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\officeSupplies\\textures\\pencil_glossyMask.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\wallClock\\textures\\wallClock.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_03.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_01.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\cork.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_brownWood.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_bluredMetal.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\dustBin\\dustBin.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\wastes\\textures\\crinkledPaper_02.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_wallPaint.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\books\\books.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\ceiling_AO.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\ceilingAirVent_AO.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\radiator\\textures\\radiator_AO.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\radiator\\radiator.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_darkWood.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\officeSupplies\\textures\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\europeMap.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\blackBoard.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\wallClock\\textures\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\desks\\desks.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\suitcase\\suitcase.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_04.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\radiator\\textures\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\coatStand\\coatStand.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\officeSupplies\\officeSupplies.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\glass.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\officeSupplies\\textures\\pencil_colorMask.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\wastes\\textures\\crinkledPaper_03.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_bareMetal.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_paper_01.png",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_02.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_woodFloor.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_leather.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\dustBin\\textures\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\woodPlanks.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_paintedPlasterWall.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\lamps\\lamps.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\classroom.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\base_scratch.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_05.jpg",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\_baseTextures\\Thumbs.db",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\wastes\\wastes.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\assets\\chairs\\chairs.blend",
        "Z:\\Host_Share_Folder\\test files\\pliki testowe\\classroom\\textures\\childDrawing\\childDrawing_07.png"
    ],
    "concent_enabled": false,
    "cost": null,
    "fee": null
}
Completed in 0.02 s
```

## Proposed Solution?
_(Optional: What could be a solution for that issue)_

**Comment by mdtanrikulu** [comment]
Golem Electron - b0.20.0 Branch - Commit hash: a29d4a863b0a5b06a38860b618a989f60246a71d

I tried to reproduce it in both testnet and mainnet, created 5 tasks on each one with given parameters, but no luck. I couldn't reproduce the issue.

**Comment by mdtanrikulu** [comment]
I found the issue and resolved here golemfactory/golem-electron@d467bf2b6c5959369b84e6b5bbd0d73574300877
The problem was changing resolution after all parameters set, writing some small number like 4 (while typing 400 i.e.), was changing max subtask limit to 0, which should min 1. 

The issue can be reproducible exactly with the given steps,

- Create task with classroom folder
- Pick master file as classroom.blend
- Set task name
- Wait for test result and accept default parameters
- Fill timeout and subtask timeout
- Go to top and change resolution y-axis as 400
- Press start task
- In task confirmation modal, subtask amount will shown as 0

Solution applied;
- Keep maximum subtask limit at minimum 1
- Don't trigger maximum subtask limit calculation function immediately while user typing, so the subtask number will be preserved until user change the resolution properly.
